### ISSUE TITLE
api design time, no xml allowed

### ISSUE THREAD

**Issue by mitechie** [bookieio/Bookie#62]
We want to have a nice API that's json/restful based. I'm not a fan of the delicious API and it's xml returns. So we need to start our own that also allows for more powerful search/filter since we're working on that.

```
/api/v1/bmarks/all
/api/v1/bmarks/search/term1/term2
/api/v1/bmarks/filter/tag1/tag2

/api/v1/tags/all
/api/v1/tags/*tags/stats
```

I think we're going to stick with the terms `filter` and `search`. 
- filter is just limited based on exact tags matches
- search is a fulltext query against the db

All paging will be done via query string. I'm not sure on paging by default or full data dump by default.

The current use case is a mobile app which should page to limit sizes sent over the wire, but as a general API we want users to have full access to their data.

**Comment by mitechie** [comment]
Started docs for API work: http://bmark.us/api.html

**Comment by mitechie** [comment]
API is started. Will add as we need. We'll need to 'freeze' v1 of the api when we released 0.3, but we can change/mess with it up until then.
